### PR TITLE
Feature/maxclassicalreg

### PR DESCRIPTION
**Pull request by cqc-melf**
add the maxnclreg predicate to the quantinuum backend

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -43,7 +43,7 @@
     packages=find_namespace_packages(include=["pytket.*"]),
     include_package_data=True,
     install_requires=[
-        "pytket >= 1.13.2, <2.0",
+        "pytket == 1.14.1rc0",
```
Comment (cqc-melf) [inline setup.py:46]: Will update this with another PR after the next release

File: pytket/extensions/quantinuum/backends/quantinuum.py [hunk 2]
```diff
@@ -261,13 +262,17 @@ def _available_devices(
     def _dict_to_backendinfo(cls, dct: Dict[str, Any]) -> BackendInfo:
         name: str = dct.pop("name")
         n_qubits: int = dct.pop("n_qubits")
+        n_cl_reg: Optional[int] = None
+        if "n_classical_registers" in dct:
```
Comment (cqc-alec) [inline pytket/extensions/quantinuum/backends/quantinuum.py:266]: Should we expect this always to be there, and error if not, as for `name` and `n_qubits`, or is it genuinely optional?

Comment (cqc-melf) [reply pytket/extensions/quantinuum/backends/quantinuum.py:266]: It is optional for the family devices( for example ("H1"), so I will keep it as it is?

Comment (cqc-melf) [reply pytket/extensions/quantinuum/backends/quantinuum.py:266]: (You can take a look on the two commits if you want to see what I tried) @cqc-melf
[make n_cl_reg required](https://github.com/CQCL/pytket-quantinuum/pull/128/commits/2fb54f5b6ff41ea71096c2c439aae6dd78c75be3)
[2fb54f5](https://github.com/CQCL/pytket-quantinuum/pull/128/commits/2fb54f5b6ff41ea71096c2c439aae6dd78c75be3) and @cqc-melf
[Revert "make n_cl_reg required"](https://github.com/CQCL/pytket-quantinuum/pull/128/commits/68a259ceb5fce0fdb8699818c501a9a003117497)
[68a259c](https://github.com/CQCL/pytket-quantinuum/pull/128/commits/68a259ceb5fce0fdb8699818c501a9a003117497)

Comment (cqc-alec) [reply pytket/extensions/quantinuum/backends/quantinuum.py:266]: OK, thanks for checking.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**